insights/parsers/tests/test_sctp.py

```diff
@@ -0,0 +1,61 @@
+import doctest
+
+import pytest
+from insights.parsers import ParseException, SkipException
+from insights.parsers import sctp
+from insights.parsers.sctp import SCTPEps
+from insights.tests import context_wrap
+
+SCTP_EPS_DETAILS = """
+ENDPT            SOCK             STY SST HBKT LPORT   UID INODE     LADDRS
+ffff88017e0a0200 ffff880299f7fa00 2   10  29   11165   200 299689357 10.0.0.102 10.0.0.70
+ffff880612e81c00 ffff8803c28a1b00 2   10  30   11166   200 273361203 10.0.0.102 10.0.0.70 172.31.1.2
+ffff88061fba9800 ffff88061f8a3180 2   10  31   11167   200 273361145 10.0.0.102 10.0.0.70
+ffff88031e6f1a00 ffff88031dbdb180 2   10  32   11168   200 273365974 10.0.0.102 10.0.0.70 192.168.11.2
+""".strip()
+
+SCTP_EPS_DETAILS_NO = """
+ENDPT            SOCK             STY SST  LPORT   UID INODE     LADDRS
+ffff88017e0a0200 ffff880299f7fa00 2   10   11165   200 299689357 10.0.0.102 10.0.0.70
+ffff880612e81c00 ffff8803c28a1b00 2   10   11166   200 273361203 10.0.0.102 10.0.0.70 172.31.1.2
+ffff88061fba9800 ffff88061f8a3180 2   10   11167   200 273361145 10.0.0.102 10.0.0.70
+ffff88031e6f1a00 ffff88031dbdb180 2   10   11168   200 273365974 10.0.0.102 10.0.0.70 192.168.11.2
+""".strip()
+
+SCTP_EPS_DETAILS_DOC = """
+ENDPT            SOCK             STY SST HBKT LPORT   UID INODE     LADDRS
+ffff88017e0a0200 ffff880299f7fa00 2   10  29   11165   200 299689357 10.0.0.102 10.0.0.70
+ffff880612e81c00 ffff8803c28a1b00 2   10  30   11166   200 273361203 10.0.0.102 10.0.0.70 172.31.1.2
+""".strip()
+
+SCTP_EPS_DETAILS_NO_2 = """
+""".strip()
+
+
+def test_sctp():
+    sctp_info = SCTPEps(context_wrap(SCTP_EPS_DETAILS))
+    assert sorted(sctp_info.sctp_local_ports) == sorted(['11165', '11166', '11167', '11168'])
+    assert sorted(sctp_info.sctp_local_ips) == sorted(['10.0.0.102', '10.0.0.70', '172.31.1.2', '192.168.11.2'])
+    assert sctp_info.sctp_eps_ips == {'ffff88017e0a0200': ['10.0.0.102', '10.0.0.70'],
+                                       'ffff880612e81c00': ['10.0.0.102', '10.0.0.70', '172.31.1.2'],
+                                       'ffff88061fba9800': ['10.0.0.102', '10.0.0.70'],
+                                       'ffff88031e6f1a00': ['10.0.0.102', '10.0.0.70', '192.168.11.2']}
+    assert len(sctp_info.search(local_port='11165')) == 1
+
+    with pytest.raises(ParseException) as exc:
+        sctp_obj = SCTPEps(context_wrap(SCTP_EPS_DETAILS_NO))
+        assert sctp_obj.sctp_eps_ips is None   # Just added ro remove flake8 warnings
+    assert 'Contents are not compatible to this parser' in str(exc)
+
+    with pytest.raises(SkipException) as exc:
+        sctp_obj = SCTPEps(context_wrap(SCTP_EPS_DETAILS_NO_2))
+        assert sctp_obj.sctp_eps_ips is None   # Just added to remove flake8 warnings
```

Comment:
@vishwanathjadhav this assert and the assert on line 47 should be to check if the object returned from SCTPEps() is `None`.  This assert is a bit confusing in that its not checking for the right thing.  If `sctp_obj.sctp_eps_ips` is empty, it would actually be an empty dict, not `None`.  The better test here would be `assert sctp_obj is None` because that is actually the expected condition that no parser is created.